pop-os/repoman
# Key management

Adds the ability to add and remove signing keys, plus view signing key information, to Repoman.

Using new APIs from Repolib, signing keys can be added, viewed, and removed from repositories using Repoman.

New features will require the `popdev:gpg-info` staging branch with updated Repolib until that PR is merged. Testing without the branch should still allow Repoman to function correctly, albeit without the updated functionality.

Testing:
* Ensure no regressions from previous functionality
* Ensure that the "Signing Key Info" tab appears in the edit dialog for repositories which have a signing key added.
* Ensure that keys for existing repositories (like `popdev-master`) can be viewed and that information matches that provided by `apt-manage`
* Ensure that the "Add signing key" button appears on sources which do not have a key
* Ensure that keys can be added to existing repositories from URLs, Fingerprints, Paths on disk, and ASCII-Armor.
* Ensure that keys added to repositories can be removed from the source.
* For fingerprint key testing, use `1E5F8BBC5BEB10AE` which is the Pop OS PPA signing key fingerprint.
* Ensure that adding a key using a Fingerprint, URL, or ASCII-Armor add a new key file in `/etc/apt/keyrings` and that the file is removed when the key is removed from the source.
* For adding a URL, use `https://raw.githubusercontent.com/pop-os/pop/master/scripts/.iso.asc`. The text at that URL can also be used for testing ASCII-Armored keys
* For adding a path, use `/etc/apt/keyrings/popdev-archive-keyring.gpg`.
* Ensure that adding a key from a Path does **not** add a new file to `/etc/apt/keyrings` and that the key file is **not** removed when the key is removed from the source.
* Keys can be test-added on a single source (i.e. the `pop-os-release` source) then removed to test another method of adding.
* Ensure that if testing without the repolib `popdev:gpg-info` branch added, none of the changes appear and that Repoman continues to function correctly.

Soft-requires pop-os/repolib#60
